### PR TITLE
FitmeUp Backend

### PR DESCRIPTION
**Pull request by barakadanny**
# Implementation of the API

In this branch, we work on :

- Database Logic and Implementation
   - Creating an ERD
- Controllers
- Models
- Validation
- Authorization 
- Authentication with JWT.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,236 @@
-# README
+<a name="readme-top"></a>
```
Comment (Meltrust) [inline README.md:1]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

- [ ] You have instructions in your readme to run tests, but I couldn't find any implemented.  Kindly add some in order to ensure your app's integrity.

- [ ] On your readme, kindly include the required `rails db:create, rails db:migrate and rails db:seed` db commands that your users will need to run in order to launch the project properly. 👍

File: app/controllers/api/v1/appointments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  def index
+    @appointments = Appointment.all
+    render json: @appointments
+  end
+
+  def show
+    @appointment = Appointment.find(params[:id])
+    render json: @appointment
+    # @appointment = Appointment.joins(:user, :trainer).find(params[:id])
+    # render json: @appointment, include: [:user, :trainer]
```
Comment (Meltrust) [inline app/controllers/api/v1/appointments_controller.rb:13]: - [ ] Kindly, remove this old commented code in order to improve your repo's cleanliness. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @barakadanny,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  API is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._

_I strongly recommend you take them into account as they can improve your final evaluation._





Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md (hunk 1 above)
Comment (skyv26) [reply README.md:1]: Removed the folder as we have not implemented any tests. `README.md` file updated according to what we have implemented.

**Review by Meltrust — APPROVED** [review]
#### Hi @barakadanny,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- The API is working well✔️ 
- Nice code organization✔️ 
- Nice routes✔️ 



Cheers and Happy coding!👏👏👏


------

**Comment by skyv26** [comment]
#### Thank you so much ❤️ for the best wishes @Meltrust .

Happy Coding 🥂

**Pull request merged**